extensions/ql-vscode/src/remote-queries/gh-api/gh-api-client.ts

```diff
@@ -0,0 +1,86 @@
+import { Credentials } from '../../authentication';
+import { OctokitResponse } from '@octokit/types/dist-types';
+import { VariantAnalysisSubmission } from '../shared/variant-analysis';
+import {
+  VariantAnalysis,
+  VariantAnalysisQueryLanguage,
+  VariantAnalysisRepoTask,
+  VariantAnalysisSubmissionRequest
+} from './variant-analysis';
+
+export async function submitVariantAnalysis(
+  credentials: Credentials,
+  submissionDetails: VariantAnalysisSubmission
+): Promise<VariantAnalysis> {
+  const octokit = await credentials.getOctokit();
```

Comment:
Would it make sense to make `octokit` an argument to the function rather than the `credentials`? I realize that we're currently using the `credentials` everywhere, but we don't really care about the actual credentials; we only need the Octokit client.

This might also make testing a bit easier since we don't need to mock this promise but can directly mock Octokit.